### ISSUE TITLE
Search engine

### ISSUE THREAD

**Issue by pierluc1** [osmandapp/Osmand#5418]
Could-you improve the search engine of Osmand on the Android version to make it powerful like Nominatim on Open Street Map? I know that I can change the search engine... But, I would like to be able to find an address that is not on a street corner without internet connection.

Tank you

**Comment by ADepic** [comment]
Which country are you in?
Because osmand only downloads addresses stored in OSM data.

However nominatim can get addresses from externel data - such as TIGER addresses in the US and ordanance survey & royal mail addresses in the UK

**Comment by pierluc1** [comment]
Canada

**Comment by vshcherb** [comment]
@lindhe I'm afraid in OSM data we have "McDonald's" same issue with nominatim btw.

**Comment by lindhe** [comment]
And that's exactly why we need a proper search engine backend which can do fuzzing, close matches etc.

**Comment by pierluc1** [comment]
A Search engine that can extrapolate the approximate position of a civic
number on a street based on the numbers at crossings.

Le dim. 22 juil. 2018 14:13, Andreas Lindhé <notifications@github.com> a
écrit :

**Comment by Christian-Git01** [comment]
I suggested to integrate Acastus-Photon app, because you get much better search results as far as I know. It is a stand-alone app (https://github.com/komoot/photon) , but it would be very comfortable to use only one app.

However, the issue was closed (https://github.com/osmandapp/Osmand/issues/9296).

**Comment by 7RST1** [comment]
I totally agree with this. The app would be so much more useful to the general public if the app supported "lazy searching".

**Comment by grravity** [comment]
I do have this issue as well with searching, even with an internet connection the results are super touchy.

Sometimes I search for a POI and it shows something with the same name like 40miles away before the one 1 mile away,  same with street names, shopping centers, etc.
If I disable internet connection, I also lose a lot of POI results, despite downloading the offline maps. It generally only shows addresses, yet even then I have the issue of touchy order of results. 

Back on the first issue with internet connection, I can type in "xx Center" and it will try to offer a place with the same name, 40miles away, before it offers the one 2 miles away. This is annoying because I have to sit there searching and try to figure out which is the correct place. Not to mention sift through all the bus stops and everything just to find the actual destination and hope I chose the correct one. If there's one thing I feel I have an issue with OSMAnd, it's searching for stuff, I hope this gets improved upon because it is something we rely on often to get around.

**Comment by sorryusernameisalreadytaken** [comment]
> 
> 
> I suggested to integrate Acastus-Photon app, because you get much better search results as far as I know. It is a stand-alone app (https://github.com/komoot/photon) , but it would be very comfortable to use only one app.
> 
> However, the issue was closed (#9296).

https://github.com/komoot/photon is really nice! If this would be integrated into Osmand it would be the last puzze to really recomend the app anybody.
